wscript

```diff
@@ -336,6 +336,8 @@ but you then may not have a local copy of the HTML manual.'''
     conf.env.append_value('CFLAGS', ['-DGTK'])
     conf.env.append_value('CXXFLAGS',
         ['-DNDEBUG', '-DGTK', '-DSCI_LEXER', '-DG_THREADS_IMPL_NONE'])
+    if conf.env['CXX_NAME'] == 'gcc' and '-O' not in ''.join(conf.env['CXXFLAGS']):
```

Comment:
Surprisingly (also for me) not - it reports gcc also for C++ compiler and interestingly gcc also when clang is used.

But maybe the whole compiler check is redundant - I just did it the same way it was done for C. Does Geany build with something else than gcc/clang? (Moreover other compilers try to have the basic flags identical to gcc)